packages/flutter_tools/lib/src/dart/analysis.dart

```diff
@@ -191,96 +197,145 @@ enum _AnalysisSeverity {
     if (diff != 0) {
       return diff;
     }
 
-    return message.compareTo(other.message);
+    return writtenError.message.compareTo(other.writtenError.message);
   }
 
   @override
   String toString() {
     // Can't use "padLeft" because of ANSI color sequences in the colorized
     // severity.
-    final String padding = ' ' * math.max(0, 7 - severity.length);
+    final String padding = ' ' * math.max(0, 7 - writtenError.severity.length);
     return '$padding${colorSeverity.toLowerCase()} $_separator '
-        '$messageSentenceFragment $_separator '
-        '${_fileSystem.path.relative(file)}:$startLine:$startColumn $_separator '
+        '${writtenError.messageSentenceFragment} $_separator '
+        '${_fileSystem.path.relative(writtenError.file)}:${writtenError.startLine}:${writtenError.startColumn} $_separator '
         '$code';
   }
 
   String toLegacyString() {
-    return '[${severity.toLowerCase()}] $messageSentenceFragment ($file:$startLine:$startColumn)';
+    return writtenError.toString();
+  }
+}
+
+/// [AnalysisError] in plain text content.
+class WrittenError {
+  WrittenError._({
+    @required this.severity,
+    @required this.type,
+    @required this.message,
+    @required this.code,
+    @required this.file,
+    @required this.startLine,
+    @required this.startColumn,
+    @required this.offset,
+  });
+
+  ///  {
+  ///      "severity":"INFO",
+  ///      "type":"TODO",
+  ///      "location":{
+  ///          "file":"/Users/.../lib/test.dart",
+  ///          "offset":362,
+  ///          "length":72,
+  ///          "startLine":15,
+  ///         "startColumn":4
+  ///      },
+  ///      "message":"...",
+  ///      "hasFix":false
+  ///  }
+  static WrittenError fromJson(Map<String, dynamic> json) {
+    return WrittenError._(
+      severity: json['severity'] as String,
+      type: json['type'] as String,
+      message: json['message'] as String,
+      code: json['code'] as String,
+      file: json['location']['file'] as String,
+      startLine: json['location']['startLine'] as int,
+      startColumn: json['location']['startColumn'] as int,
+      offset: json['location']['offset'] as int,
+    );
   }
+
+  String severity;
+  String type;
+  String message;
+  String code;
+
+  String file;
+  int startLine;
+  int startColumn;
+  int offset;
+
+  static final Map<String, _AnalysisSeverity> _severityMap =
+      <String, _AnalysisSeverity>{
+    'INFO': _AnalysisSeverity.info,
+    'WARNING': _AnalysisSeverity.warning,
+    'ERROR': _AnalysisSeverity.error,
+  };
+
+  _AnalysisSeverity get severityLevel =>
+      _severityMap[severity] ?? _AnalysisSeverity.none;
+
+  String get messageSentenceFragment {
+    if (message.endsWith('.')) {
+      return message.substring(0, message.length - 1);
+    } else {
+      return message;
+    }
+  }
+
+  @override
+  String toString() {return '[${severity.toLowerCase()}] $messageSentenceFragment ($file:$startLine:$startColumn)';}
```

Comment:
formatting: if you use `{ }` for a method body, its preferable to start the actual implementation on the next line. Only use `=>` for methods on the same line.